cpp/src/arrow/compute/row/grouper.h

```diff
@@ -30,8 +30,15 @@
 namespace arrow {
 namespace compute {
 
-/// \brief A segment of contiguous rows for grouping
-struct ARROW_EXPORT GroupingSegment {
+/// \brief A segment.
+/// A segment group is a chunk of continous rows that has the same segment key. (For
```

Comment:
It's a bit confusing to start with a description of a segment group when the docstring is for segment. I'd say here something like "A segment is a chunk of contiguous rows that have the same segment key. Multiple contiguous segments having a common segment key form a segment group.".

Note that there is a typo: "that has" should be "that have".

Reply:
I think it doesn't really make sense to talk about segment without talking about segment group first - segment group is the top-level concept, segment is how we track batches within a segment group